lib/elixir/lib/string.ex

```diff
@@ -1915,6 +1914,9 @@ defmodule String do
       iex> String.to_float("3.0")
       3.0
 
+      String.to_float("3")
```

Comment:
That's what I originally had in mind. The downside is it generates annoying warning when running tests because the compiler recognizes it will fail.